Fix how FindNetCDF finds MPI

kgerheiser opened the pull request:
Search only for given components and then link to corresponding NetCDF library

I was able to build wgrib2 and w3emc with these changes

kgerheiser committed:
Fix FindNetCDF with MPI

Search only for COMPONENTS passed to FindNetCDF

kgerheiser committed:
Check if target exists before linking

kgerheiser commented:
Ah, but what if no components are specified? It will be empty when it should be `C`? Am I correct @aerorahul 

Let me see

kgerheiser committed:
Search for enabled languages

aerorahul commented on Modules/FindNetCDF.cmake:
```diff
@@ -178,11 +178,16 @@ if( NOT _val MATCHES "^(yes|no)$" )
+
+if(NetCDF_PARALLEL)
+  # Searches for all enabled languages
+  find_package(MPI)
```
Shouldn't there be REQUIRED here. Otherwise, it is just going to fail elsewhere when linking and we will have to scratch our heads (well, not us!)
May be have an error trap, 
```
if(NOT MPI_FOUND)
  message(FATAL_ERROR "NetCDF has parallel enabled, but was unable to find MPI")
endif()
```

aerorahul commented on Modules/FindNetCDF.cmake:
```diff
@@ -242,11 +247,9 @@ foreach( _comp IN LISTS _search_components )
-          target_link_libraries(NetCDF::NetCDF_C INTERFACE MPI::MPI_C)
-        endif()
+      endif()
+      if(TARGET MPI::MPI_${_comp})
```
I don't like the `TARGET` here.
Can we use:
```
if(MPI_${_comp}_FOUND)
```
Or
```
if(NETCDF_PARALLEL)
```

aerorahul reviewed (changes requested):
thoughts?

kgerheiser committed:
Search for MPI from NetCDF components

If no components are given, default to C.

Add checks if the lagnuage is enabled.

kgerheiser commented:
Fixed.

What do you think of what I added? I went back to checking for the NetCDF components instead of all the enabled languages and added some checks. It seems better to search MPI using that than all the enabled languages.

Or am I over-complicating it?

kgerheiser committed:
Typo

kgerheiser committed:
Typo

kgerheiser committed:
Make it simple again

aerorahul commented on Modules/FindNetCDF.cmake:
```diff
@@ -178,11 +178,15 @@ if( NOT _val MATCHES "^(yes|no)$" )
   set(NetCDF_PARALLEL FALSE CACHE STRING "NetCDF has no parallel IO capability." FORCE)
 endif()
 
+
```
extra line.

kgerheiser committed:
Remove extra space

aerorahul reviewed (approved):
looks good.
This is much cleaner.
I tested it with clang and mpich on macOS and w3emc built.
I also notice that the MPICH include path is included in the build log of w3emc, so that is a confirmation.
